### ISSUE TITLE
Allow custom configs for CKEditor

### ISSUE THREAD

**Issue by mtnorthrop** [Kunstmaan/KunstmaanBundlesCMS#402]
Would it be possible to allow custom configs to be set for the CKEditor in the Admin bundle?

It is already possible to set a custom toolbar config by overriding `KunstmaanAdminBundle:Default:_ckeditor_configs.html.twig`, however, I'd like to be able to override the `CKEDITOR.config` object properties as well, using an external config file.

The CKEditor documentation describing how to do this is here: http://docs.ckeditor.com/#!/guide/dev_configuration

Maybe this could be done in a way similar to the way the custom toolbar settings are loaded. 

The `customConfig` property could be added to the in-page settings of the `CKEDITOR.replace()` call in `AdminBundle/Resources/ui/js/_rich-editor.js`, and could be assigned to a variable that is an empty string by default:

```javascript

        customConfig = customConfigFile || ''; // Get URL set in customConfigFile if set

        // Place CK
        CKEDITOR.replace(elId, {
            skin: 'bootstrapck',
            startupFocus: false,
            height: elHeight,
            bodyClass: 'CKEditor',

            filebrowserBrowseUrl: fileBrowseUrl,
            filebrowserWindowWidth: 970,

            filebrowserImageBrowseUrl: imageBrowseUrl,
            filebrowserImageBrowseLinkUrl: imageBrowseUrl,
            filebrowserImageWindowWidth: 970,

            filebrowserImageUploadUrl: '',

            enterMode: elEnterMode,
            shiftEnterMode: elShiftEnterMode,

            toolbar: elToolbar,
            customConfig: customConfig // Include custom config file setting
        });
```

This way, the user would have the option of setting the url for their custom config file by overriding the `KunstmaanAdminBundle:Default:_ckeditor_configs.html.twig` file and setting the `customConfigFile` variable there.

Not only would this give users access to all the CKEditor config settings, but would also allow them to load external CKEditor plugins (something that I need for my current project).

Would something like this be possible?

**Comment by denbatte** [comment]
Hi  mtnorthrop,
There was a issue doing this in 3.1.0 but has been fixed in the 3.1.1 version.
https://github.com/Kunstmaan/KunstmaanBundlesCMS/pull/375

**Comment by mtnorthrop** [comment]
Hello denbatte,

Thanks for your response. 

The changes in #375 only allowed customization of the CKEditor toolbar, but I'm looking for a way to have access to all of the CKEditor config settings and the only way to do that, as far as I know, is to set the `customConfig` property when the editor is placed on the page with `CKEDITOR.replace()`.

So this is a separate issue from the one that was fixed in #375.

I'll try to fork the bundle and submit a pull request that adds the option to provide an external config file for CKEditor.

**Comment by mtnorthrop** [comment]
The PR that fixes this issue has been merged into the master branch, so I'll close this ticket.
